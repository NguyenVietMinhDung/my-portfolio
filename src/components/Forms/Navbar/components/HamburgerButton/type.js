export type Props = {
  closeIconUrl: string,
  hamburgerIconUrl: string,
  isVisible: boolean,
  openNavigationMenu: Function,
};
