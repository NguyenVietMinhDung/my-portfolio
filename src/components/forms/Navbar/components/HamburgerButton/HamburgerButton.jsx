// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './HamburgerButton.scss';
import { Props } from './type';

const getMenuIconUrl = (
  isVisible,
  closeIconUrl,
  hamburgerIconUrl,
) => (isVisible ? closeIconUrl : hamburgerIconUrl);

const HamburgerButton = (props: Props) => {
  const {
    isVisible, closeIconUrl, hamburgerIconUrl, openNavigationMenu,
  } = props;
  return (
    <img
      styleName="hamburger-btn"
      src={getMenuIconUrl(isVisible, closeIconUrl, hamburgerIconUrl)}
      alt=""
      onClick={openNavigationMenu}
    />
  );
};

export default memo<Props>(CSSModules(HamburgerButton, styles));
