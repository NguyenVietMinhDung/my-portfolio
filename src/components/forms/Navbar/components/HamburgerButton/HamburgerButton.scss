@import '../../../../../styles/abstracts/variables';

.hamburger-btn {
	width: 35px;
	height: 35px;

	@media (min-width: $layout-breakpoint) {
		display: none;
	}

	@media (max-width: $layout-breakpoint) {
		display: flex;
	}
}
