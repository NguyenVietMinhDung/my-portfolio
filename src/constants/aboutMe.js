export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const MAX_SKILL_VALUE = 100;
