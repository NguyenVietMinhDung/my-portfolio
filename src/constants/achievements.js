export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
