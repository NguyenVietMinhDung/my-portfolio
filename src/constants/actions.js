export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
