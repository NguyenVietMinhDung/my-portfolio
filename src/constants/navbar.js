export const items = [
  {
    itemId: 'header',
    name: 'Home',
    url: '#home',
  },
  {
    itemId: 'about-me',
    name: 'About',
    url: '#about',
  },
  {
    itemId: 'work-experiences',
    name: 'Experiences',
    url: '#experiences',
  },
  {
    itemId: 'relevant-projects',
    name: 'Projects',
    url: '#projects',
  },
  {
    itemId: 'my-educations',
    name: 'Educations',
    url: '#educations',
  },
  {
    itemId: 'my-achievements',
    name: 'Achievements',
    url: '#achievements',
  },
  {
    itemId: 'footer',
    name: 'Contact',
    url: '#contact',
  },
];
export hamburgerIconUrl from '../assets/images/hamburger-icon.png';
export closeIconUrl from '../assets/images/close-icon.png';
