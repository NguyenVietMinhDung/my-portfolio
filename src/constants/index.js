import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
