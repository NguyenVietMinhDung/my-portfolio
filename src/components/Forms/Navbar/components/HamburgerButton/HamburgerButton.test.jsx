import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { closeIconUrl, hamburgerIconUrl } from '../../../../../constants/navbar';
import HamburgerButton from './HamburgerButton';

const defaultProps = {
  closeIconUrl,
  hamburgerIconUrl,
  openNavigationMenu: jest.fn(),
  isVisible: false,
};

describe('<HamburgerButton />', () => {
  test('renders correctly', () => {
    const component = shallow(<HamburgerButton {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  test('handle click event', () => {
    const component = mount(<HamburgerButton {...defaultProps} isVisible />);
    component.find('img').simulate('click');
    expect(component.props().openNavigationMenu).toBeCalled();
  });
});
