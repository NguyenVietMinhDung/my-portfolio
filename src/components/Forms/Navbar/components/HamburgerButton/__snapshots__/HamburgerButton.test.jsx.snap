// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<HamburgerButton /> renders correctly 1'] = `
<img
  alt=""
  className="hamburger-btn"
  onClick={[MockFunction]}
  src="test-file-stub"
/>
`;
